02.calendar/calendar.rb

```diff
@@ -8,36 +8,34 @@
 year = params["y"] ? params["y"].to_i : today.year
 
 unless month.between?(1, 12)
-  puts "cal: 13 is neither a month number (1..12) nor a name"
+  puts "cal: #{params["m"]} is neither a month number (1..12) nor a name"
   return
 end
 
 unless year.between?(1, 9999)
-  puts "cal: year `10000' not in range 1..9999"
+  puts "cal: year `#{params["y"]}' not in range 1..9999"
   return
 end
 
-first_day = Date.new(year, month, 1)
-last_day = Date.new(year, month, -1)
-beginning_of_month = first_day.wday
+first_date = Date.new(year, month, 1)
+last_date = Date.new(year, month, -1)
+first_day_of_month = first_date.wday
 
 puts ("#{month}月 #{year}").center(20)
 puts "日 月 火 水 木 金 土"
 
-space = format("%3s", "")
-print space * beginning_of_month
+print "   " * first_day_of_month
 
-(first_day..last_day).each do |day|
-  if day == today
+(first_date..last_date).each do |date|
+  if date == today
     print "\e[30m\e[47m"
-    printf("%2s", day.day.to_s)
+    printf("%2s", date.day.to_s)
     print "\e[0m"
-    print " "
   else
-    printf("%2s", day.day.to_s)
-    print " "
+    printf("%2s", date.day.to_s)
   end
-  if day.saturday?
+  print " "
+  if date.saturday?
     print "\n"
```

Comment:
37行目をなくして、こうした方がいいかも？

```suggestion
    print "\n"
  else
    print " "
```

でないと、土曜日のうしろに無駄なスペースが一つ入ってしまうので。